packages/common/package-utils.js

```diff
@@ -0,0 +1,32 @@
+import { mkdirSync, writeFileSync } from 'fs';
+import _ from 'lodash';
+
+/**
+ * Get the distributable package manifest ready for publishing.
+ *
+ * @param {'esm' | 'cjs'} format
+ */
+export const getDistPackageManifest = (pkg, format = 'esm') => {
+  const distPkg = _.cloneDeep(pkg);
```

Comment:
Mhh, we could probably use [spread operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax). We are not mutating any deep values. Also we could use destructor here

```JS
const { devDependencies, scripts, dependencies, ...distPkg } = pkg;
```

And later when we remove `lodash-es` we can delete it from `dependencies`. And when returning the package
```JS
return {
  ...distPkg,
  dependencies
}
```

Reply:
@karelhala Note that I've removed this file in a later update.

`main` field of package manifest is now set via `publishConfig`, other fields like `files` (which have no impact when importing from within another monorepo package) are kept as before.

`types` field was removed. When absent, TypeScript checks the `main` field and, if present, infers the `.d.ts` from it, see [here](https://www.typescriptlang.org/docs/handbook/declaration-files/dts-from-js.html#editing-the-packagejson) for reference.

> `“main”:“index.js”` :arrow_right: `index.d.ts`

As for Lodash, `lib-webpack` is built against `lodash-es` since the standard CommonJS `lodash` version has tons of issues when it comes to tree shaking and module bundling in general.

At the end of the build, we simply replace the `require()` for `lodash-es` with `lodash` in the generated JS bundle.